### ISSUE TITLE
Memory leak in the function argument passed as a binary operation

### ISSUE THREAD

**Issue by Thirumalai-Shaktivel** [lfortran/lfortran#2627]
```fortran
program example
    implicit none
    real :: R(4), V(4)
    print *, func(R, V, 2)
contains
    integer function func(R, V, l) result(res)
        real, intent(in) :: R(:), V(:)
        integer, intent(in) :: l
        res = find_ctp(V + l*(l+1)/(2*R**2))
    end function

    integer function find_ctp(V) result(ctp)
        real, intent(in) :: V(:)
    end function
end program example
```
```
...
leaks Report Version: 4.0, multi-line stacks
Process 54867: 189 nodes malloced for 14 KB
Process 54867: 2 leaks for 64 total leaked bytes.

STACK OF 1 INSTANCE OF 'ROOT LEAK: <malloc in _lfortran_malloc>':
3   expr2.out                             0x104a4fea8 main + 68
2   expr2.out                             0x104a4fd5c func_r_v + 704
1   liblfortran_runtime.0.20.3.dylib        0x104eec124 _lfortran_malloc + 24  lfortran_intrinsics.c:1550
0   libsystem_malloc.dylib                0x188e38a44 _malloc_zone_malloc_instrumented_or_legacy + 136 
====
    1 (32 bytes) ROOT LEAK: <malloc in _lfortran_malloc 0x143704180> [32]

STACK OF 1 INSTANCE OF 'ROOT LEAK: <malloc in _lfortran_malloc>':
3   expr2.out                             0x104a4fea8 main + 68
2   expr2.out                             0x104a4fb70 func_r_v + 212
1   liblfortran_runtime.0.20.3.dylib        0x104eec124 _lfortran_malloc + 24  lfortran_intrinsics.c:1550
0   libsystem_malloc.dylib                0x188e38a44 _malloc_zone_malloc_instrumented_or_legacy + 136 
====
    1 (32 bytes) ROOT LEAK: <malloc in _lfortran_malloc 0x143704160> [32]
...
```

**Comment by czgdp1807** [comment]
I will work on it on Monday.

**Comment by Thirumalai-Shaktivel** [comment]
Here is a simple one:
```fortran
program example
    implicit none
    integer :: x(3), y(4)
    x = 45
    y = 23
    print *, func(x, y)
contains
    integer function func(x, y)
        integer, intent(in) :: x(:), y(:)
        print *, sum(x*y)
    end function func
end program example
```

**Comment by czgdp1807** [comment]
This will be fixed in #2615. I will do this on Monday.
